s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/query/QueryText.java

```diff
@@ -0,0 +1,50 @@
+package pl.cyfronet.s4e.data.repository.query;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class QueryText extends QueryDecorator {
+    public QueryText(QueryBuilder queryBuilder) {
+        super(queryBuilder);
+    }
+
+    public StringBuilder prepareQueryAndParameters(Map<String, Object> params, List<Object> parameters, StringBuilder resultQuery) {
+        super.prepareQueryAndParameters(params, parameters, resultQuery);
+        queryBySatellitePlatform(params, parameters, resultQuery);
+        queryBy("productType", params, parameters, resultQuery);
+        queryBy("polarisation", params, parameters, resultQuery);
+        queryBy("processingLevel", params, parameters, resultQuery);
+        queryBy("sensorMode", params, parameters, resultQuery);
+        queryBy("collection", params, parameters, resultQuery);
+        queryBy("timeliness", params, parameters, resultQuery);
+        queryBy("instrument", params, parameters, resultQuery);
+        queryBy("productLevel", params, parameters, resultQuery);
+        queryBy("relativeOrbitNumber", params, parameters, resultQuery);
+        queryBy("absoluteOrbitNumber", params, parameters, resultQuery);
+        return resultQuery;
+    }
+
+    private void queryBySatellitePlatform(Map<String, Object> params, List<Object> parameters, StringBuilder resultQuery) {
+        if (params.containsKey("satellitePlatform")) {
+            // [string] satellitePlatform -> spacecraft
+            resultQuery.append(" AND metadata_content->>'spacecraft' = ? ");
+            parameters.add(String.valueOf(params.get("satellitePlatform")));
```

Comment:
Add a test that verifies that SQL injection cannot be done exploiting methods such as this.

Reply:
Okay, I missed the SearchServiceTest::testSqlInjection